### DISCUSSION TITLE
Welcome to 3DPrintColorizer Discussions!

### DISCUSSION THREAD

**Discussion by Sakati84** [Sakati84/3DPrintColorizer#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by csanii123** [comment]
I saw your idea on hackaday.com and thought it was great. I have a Creality Ender 3 v2 with a stock config and it is printing standard prints very well. Since I don't want to "break" my standard prints I designed a modular "snap-on" pen holder for the fan enclosure so I can add/remove the colorizer without having to do any changes to the print mechanism. I am going to try my 1st dual color print in the next few days as I still have the pen rack to print and adjust. I did have a question for you regarding getting multicolor STL files. I wanted to just take 2 standard shapes (cube+cylinder) and print them merged as a 2 color print. I loaded them into Cura as 2 shapes and assigned the extruders to them but when I merge it shifts the 2 so that they are inside each other and not in the positions I wanted. How do I do the merge so that the end shapes end up where I want them to ?

**Reply by Sakati84** [reply]
You need to Design them in your CAD Program so that they are correctly positioned. In your case here you can just skip merging them. This is only important when you have special models that are designed for multi color printing and makes it easier to align then correctly.

**Comment by csanii123** [comment]
Ok thx for the confirmation. I was suspecting that.  My E3v2 "snap-on" allows me to use all the parts you designed except the PenHolderArm since that won't fit on an Ender 3 v2. This way all "standard" except the Ender 3 v2 "Snap-on". Will send you a picture of the test print when done...hopefully this works as well as I hope.

**Reply by Sakati84** [reply]
I uploaded an Ender 3v2 PenHolderArm some hours ago. So you can also use this if you want.

**Reply by csanii123** [reply]
Just checked as I thought maybe you were joking....Lots of items just uploaded but all BLtouch related and I think all require that I undo the wheel screw ...which I want to avoid. I like my clip-on idea for easy add/remove and no impact to my currently very successful prints with my stock machine. So far 20+ prints without issues and no relevelling. Let me test my clip-on and if it works I'll send to you so you can post it for those without a BLtouch or those not wanting to adjust screws.

**Reply by Sakati84** [reply]
Would love to see your Design when you tested it!

**Reply by csanii123** [reply]
Prototype#1 worked better than I expected! I got it to pick up the pen but I must have my settings off a little as I had to manually put the pen back. I figured out a bit too late is that with interlace on you have to make sure you have an odd number of layers or else your last layer will be white. At least that is easy to fix. The other item which may be my settings or my adapter is that the pen was slightly off where to color. I'll have to play with this some more...very promising though....Attached are some of the pics I took...and yes that is tape connecting my adapter to your pen holder. I figured later I will glue together or add a spot to use the screw holes. I also made a single Penholder as I wanted to make sure this worked before printing a 6 pen holder :)
![Ender3v2Adapter](https://user-images.githubusercontent.com/67832530/128071673-6f05898a-41b4-40d5-a192-cc87c46693d1.jpg)
![SinglePenRack](https://user-images.githubusercontent.com/67832530/128071676-64b27780-868d-4714-9322-092671056867.jpg)
![Cylinder](https://user-images.githubusercontent.com/67832530/128071679-b8172484-f46e-4af7-b189-918044a5b7fb.jpg)

**Reply by Sakati84** [reply]
Hey, looks good for a first try. You need to use transparent PLA to get the best results. This will also solve the problem with the last layer being white.

**Reply by csanii123** [reply]
I never thought to use transparent. I was thinking white would show up the colors the best. It figures... I bought a bunch of different filament...but none transparent since I didn't think I would use it. :)

**Reply by Sakati84** [reply]
It's way better with transparent PLA. You can buy the cheap Amazon Basics Pla - link is in the Wiki. If you buy a diffrent one: the cheaper the better!

**Reply by csanii123** [reply]
Thanks for the tip. An idea I thought of since most things I print are small is to have a crossbeam lower down and mount the penholder to that. I was building a laser cutter with 20x20 extrusions so may borrow an extrusion to give that a try. This way less vertical travel needed.

**Reply by csanii123** [reply]
I ordered some transparent filament that should arrive Friday so I can test out this weekend. I also found that my pen holder  cracked so that might be why the alignment was off. I think I will print another pen holder but scale it up by 5% so it doesn't crack. Attached are my STL files for prototype#1.
[Cyrus_E3v2_MulticolorAdapter_Files.zip](https://github.com/Sakati84/3DPrintColorizer/files/6934248/Cyrus_E3v2_MulticolorAdapter_Files.zip)

Did you see the email from Creality ?

**Reply by Sakati84** [reply]
I saw the mail. I have just hidden the message so the mail address is not public anymore. We are in contact but nothing has really come out of it yet.

**Reply by csanii123** [reply]
Ok. Just wanted to make sure you saw it. Hopefully something good comes of it :)

**Comment by Creality-mandy** [comment]
Hi, this is Creality Mandy, I saw your great work about Ender 3. I want to cooperate with you to do more interesting thing. If possible, please contact me by email( zhujinyun@creality.com ).Looking forward to hearing from you. Thanks.

**Comment by ssilverm** [comment]
Just saw Teaching Tech uploaded a video :)

https://www.youtube.com/watch?v=UnWxbU7Hfro

**Comment by DatMinh94** [comment]
Hello @Sakati84. I'm a doctor and I need to print colorized anatomical model for surgical planning at my hospital. We don't have a big budget to buy a polyjet printer, so your idea is our very savior. We are trying to use dual extruder printer. One nozzle will do the color printing using your genious method and the other nozzle will print soluble support. Can you modify the plug in so that the last slot or "virtual nozzle" in Cura is always reserved for the second nozzle to print PVA? We very much appreciate your job. That would be perfect if we can print PVA and multicolor with a FDM machine.

**Comment by DatMinh94** [comment]
![CreatBot_DX_DX_Plus_Description_11-1000x1000-270x270](https://user-images.githubusercontent.com/78602717/154334921-7d0fb004-eba5-4446-a09b-4f924e4b255f.jpg)
